packages/xarc-app/src/index.ts

```diff
@@ -0,0 +1,207 @@
+const logger = require("../lib/logger");
+
+let AppMode;
+
+function getAppMode() {
+  if (AppMode) {
+    return AppMode;
+  }
+
+  try {
+    const archetype = require("@xarc/app-dev/config/archetype")();
+    return (AppMode = archetype.AppMode);
+  } catch (e) {
+    return (AppMode = makeAppMode(constants.PROD_DIR));
+  }
+}
+
+/**
+ * Load CSS module run time hook
+ *
+ * @param options - options for the hook
+ */
+export function cssModuleHook(
+  options: {
+    /**
+     * template for scope name to generate, default: "[hash:base64]"
+     */
+    generateScopedName?: string;
+    /**
+     * rootDir - src or lib in production
+     */
+    rootDir?: string;
+    /**
+     * any other options for css-modules-require-hook
+     */
+    [key: string]: any;
+  } = {}
+) {
+  const defaultRootDirPath = process.env.NODE_ENV === "production" ? "lib" : "src";
+  options.generateScopedName = options.generateScopedName || "[hash:base64]";
+  options.rootDir = options.rootDir || Path.resolve(defaultRootDirPath);
+
+  require("css-modules-require-hook")(options);
+}
+
+/**
+ * Load the require hook to support isomorphic assets when doing SSR
+ */
+export function isomorphicExtendRequire() {
+  return isoExtRequire({
+    processAssets: (assets: any) => {
+      const appSrcDir = (getAppMode().getEnv() || getAppMode().lib.dir).split("/")[0];
+      if (appSrcDir !== getAppMode().src.dir && assets.marked) {
+        const marked = assets.marked;
+        Object.keys(marked).forEach(k => {
+          if (k.startsWith(getAppMode().src.client) || k.startsWith(getAppMode().src.server)) {
+            const nk = k.replace(getAppMode().src.dir, appSrcDir);
+            marked[nk] = marked[k];
+          }
+        });
+      }
+
+      return assets;
+    }
+  }).catch((err: Error) => {
+    isoExtRequire._instance.interceptLoad();
+  });
+}
+
+/**
+ * Load run time supports
+ *
+ * @param options - options on what to load
+ * @param callback - callback after loaded (returns Promise if no callback)
+ * @returns nothing
+ */
+export function load(
+  /**
+   * support load options or a callback for when support load is done
+   */
+  options:
+    | {
+        /**
+         * - boolean: true to load @babel/register
+         * - Object: options to be passed to @babel/register
+         */
+        babelRegister?: any;
+        /**
+         * if true, then load and setup CSS module runtime for node.js
+         */
+        cssModuleHook?: boolean;
+        /**
+         * if no CSS module hook, then a default ignore hook is load to avoid errors
+         * when trying to load CSS modules.
+         *
+         * Set this to false to avoid loading the ignore hook.
+         */
+        ignoreStyles?: boolean;
+        /**
+         * Set to false to avoid loading node.js require hook to handle isomorphic assets
+         */
+        isomorphicExtendRequire?: boolean;
+      }
+    | Function,
+  callback?: Function
```

Comment:
unsure who may use this, but do we have any extra info on this callback? maybe it's actually `(config: Config) => void`? not saying it is, just curious though!